src/components/ec-currency-input/ec-currency-input.vue

```diff
@@ -0,0 +1,189 @@
+<template>
+  <div class="ec-currency-input">
+
+    <label
+      v-if="label || note"
+      class="ec-currency-input__label"
+      :for="id"
+    >
+      <span
+        v-if="label"
+        data-test="ec-currency-input__label-text"
+        class="ec-currency-input__label-text"
+      >{{ label }}</span>
+
+      <span
+        v-if="note"
+        data-test="ec-currency-input__note"
+        class="ec-currency-input__note"
+      >{{ note }}</span>
+    </label>
+
+    <div
+      ref="popperWidthReference"
+      class="ec-currency-input__input-group"
+    >
+      <ec-dropdown
+        v-model="currencyModel"
+        :class="{'ec-currency-input__currencies--is-focused': currenciesHasFocus}"
+        :id-from-parent="id"
+        :items="currenciesItems"
+        :popper-modifiers="popperModifier"
+        :popover-options="popoverOptions"
+        class="ec-currency-input__currencies"
+        is-in-group="right"
+        is-search-enabled
+        :error-message="errorMessage"
+        @focus="currenciesHasFocus = true"
+        @blur="currenciesHasFocus = false"
+      />
+      <ec-amount-input
```

Comment:
You have to pass selected currency from currencyModel into this component too. Amount input relies of currency. If currency doesn't have decimals, it will not allow you to type decimals.